src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -683,6 +685,23 @@ private void ProcessGoAwayFrame(FrameHeader frameHeader)
             _incomingBuffer.Discard(frameHeader.Length);
         }
 
+        internal async Task FlushAsync() // no cancellation token: we don't want individual operation tokens applying to the whole connection
```

Comment:
We certainly don't want a cancellationToken from the request stream to apply to the whole connection, i.e. FlushOutgoingBytesAsync. But we could apply it to acquiring the write lock. I don't think it's a big deal as the write lock shouldn't be held that long, but it seems worth considering.